.github/workflows/test_dbt_models.yaml

```diff
@@ -22,8 +22,63 @@ jobs:
         with:
           role-to-assume: ${{ secrets.AWS_IAM_ROLE_TO_ASSUME_ARN }}
 
-      - name: Test models
-        run: dbt test --target "$TARGET"
+      - name: Install Python requirements
+        run: pip install -r ./.github/scripts/requirements.transform_dbt_test_results.txt
+        shell: bash
+
+      - name: Run tests
+        run: |
+          # dbt doesn't differentiate between test failures and errors, but we
+          # need to since we expect failures. Do this by capturing the output
+          # and checking its contents to look for errors
```

Comment:
This is annoying but it works, as evidenced by [this workflow run](https://github.com/ccao-data/data-architecture/actions/runs/8527517308/job/23359117365) that failed due to the QC schema not existing for my branch.